lib/posix/options/key.c

```diff
@@ -127,19 +128,21 @@ int pthread_key_create(pthread_key_t *key,
  */
 int pthread_key_delete(pthread_key_t key)
 {
+	int ret;
 	size_t bit;
-	__unused int ret;
 	pthread_key_obj *key_obj;
 	struct pthread_key_data *key_data;
 	sys_snode_t *node_l, *next_node_l;
-	k_spinlock_key_t key_key;
 
-	key_key = k_spin_lock(&pthread_key_lock);
+	ret = sys_sem_take(&pthread_key_lock, K_FOREVER);
```

Comment:
It's probably obvious and I just can't see it: Why don't you use your new contruct here and in the other cases below?

Reply:
Done - it was mainly just travelling, being busy, and laziness. I didn't quite get a chance to fully look at all of the cases until this morning :-)